services/src/main/java/org/keycloak/protocol/oidc/TokenManager.java

```diff
@@ -519,16 +519,14 @@ public AccessToken transformUserInfoAccessToken(KeycloakSession session, AccessT
         for (ProtocolMapperModel mapping : mappings) {
 
             ProtocolMapper mapper = (ProtocolMapper)sessionFactory.getProviderFactory(ProtocolMapper.class, mapping.getProtocolMapper());
-            if (mapper == null || !(mapper instanceof OIDCAccessTokenMapper)) continue;
 
-            if(mapper instanceof UserInfoTokenMapper){
-                token = ((UserInfoTokenMapper)mapper).transformUserInfoToken(token, mapping, session, userSession, clientSession);
-                continue;
+            if (mapper instanceof UserInfoTokenMapper) {
+                token = ((UserInfoTokenMapper) mapper).transformUserInfoToken(token, mapping, session, userSession, clientSession);
+            } else if (mapper instanceof OIDCAccessTokenMapper) {
```

Comment:
Shall I remove the fallback to OIDCAccessTokenMapper? Without it some tests in the integration-test/base fail...
e.g. {{AccessTokenTest.*}} and {{UserInfoTest.*}}.

In {{org.keycloak.testsuite.util.UserInfoClientUtil#testSuccessfulUserInfoResponse}} the check `Assert.assertEquals(expectedEmail, userInfo.getEmail());` fails because `userInfo.getEmail()` is `null`.

Which mapper contributes the email?